e2e_test/combined_test.go

```diff
@@ -0,0 +1,82 @@
+package e2e_test
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCombined(t *testing.T) {
+	// combined tests combine multiple resources and can thus not be run in parallel
+	serverID := createServer(t, "test-server", "cpx11", "ubuntu-24.04")
```

Comment:
Can we move constants like the server type and image to variables? This makes them easier to replace if the resource is deprecated/removed in the future or we want to use something else for better utilization

Similar to how its done in Terraform: https://github.com/hetznercloud/terraform-provider-hcloud/blob/ac90449bdd4782e0c58930b638cb3e8f4e313725/internal/teste2e/testing.go#L19-L43